.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,85 @@
+name: Unit Tests
+
+on: pull_request
+
+jobs:
+  tests:
+    name: "WP: ${{ matrix.wp_version }} - PHP: ${{ matrix.php }} - ES: ${{ matrix.es_version }} (MU: ${{ matrix.multisite }})"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # do not fail fast, let all the failing tests fail.
+      matrix:
+        php: [8.2, 8.3, 8.4]
+        es_version: [7.17.25, 8.16.1]
+        multisite: [0]
+        wp_version: ["latest", "nightly"]
+    env:
+      CACHEDIR: /tmp/test-cache
+      WP_CORE_DIR: /tmp/wordpress/
+      WP_TESTS_DIR: /tmp/wordpress-tests-lib
+      WP_VERSION: ${{ matrix.wp_version }}
+      WP_MULTISITE: ${{ matrix.multisite }}
+    services:
+      mysql:
+        image: mysql:5.7
```

Comment:
is it worth testing the new mysql version? As far as I know, VIP doesn't use this version anymore.

Maybe add to the matrix?

Reply:
Added a matrix of newer versions!